jrdalino/myproject-customer-service-python
# Feature/add unittesting

Changes: 
- Flask application
  - decouple create_app
  - decouple db client
  - fix module visibility issues
  - minor changes on errorhandlers
  - add requirements.txt and additional required packages
  - change data return formatting to the ff example: 
    ```json
    {
      "customer" : {
        "data": {
           "customer_id": "2b473002-36f8-4b87-954e-9a377e0ccbec",
           "first_name": "Emelyne",
           "last_name": "Plumley",
           "email": "eplumley1@wikipedia.org",
           "dob" : "January 1, 2000",
           "gender" : "Female"         
       },
        "status": "CREATED OK"
      }
    }
    ```

- Testing - Add unit testing
  - add unit testing for application health check (pytest-flask)
  - add unit testing for customer_table_crud (moto, unittest, pytest)
      - test_get_all_customer()
     -  test_get_customer()
      - test_create_customer()
     -  test_update_customer()
      - test_delete_customer()
  - add unit testing for db client

To run tests run the ff command in the root folder of this project:
```bash
$ pytest
```

additional changes:
- code clean up
- add customer extra fields into `customer_routes.py`
- clean up db.py
- load test data from json file in setup for `test_customer_routes.py`
- remove 404 error handler
